staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -49,11 +61,167 @@ func profileLegacy(pod *corev1.Pod, containerName string, target runtime.Object)
+						},
+					}
+					if c.SecurityContext == nil {
+					} else if c.SecurityContext.Capabilities == nil {
+						pod.Spec.EphemeralContainers[i].SecurityContext.Capabilities = &corev1.Capabilities{
+							Add: []corev1.Capability{"SYS_PTRACE"},
+						}
+					} else {
+						pod.Spec.EphemeralContainers[i].SecurityContext.Capabilities.Add = append(c.SecurityContext.Capabilities.Add, "SYS_PTRACE")
+					}
+				}
+			}
+		} else {
+			// For copy of pod: sets SYS_PTRACE in debugging container, sets shareProcessNamespace
+			pod.Spec.ShareProcessNamespace = pointer.BoolPtr(true)
+			for i, c := range pod.Spec.Containers {
+				if c.Name == containerName {
+					pod.Spec.Containers[i].SecurityContext = &corev1.SecurityContext{
+						Capabilities: &corev1.Capabilities{
+							Add: []corev1.Capability{"SYS_PTRACE"},
+						},
+					}
+					if c.SecurityContext == nil {
+					} else if c.SecurityContext.Capabilities == nil {
+						pod.Spec.Containers[i].SecurityContext.Capabilities = &corev1.Capabilities{
+							Add: []corev1.Capability{"SYS_PTRACE"},
+						}
+					} else {
+						pod.Spec.Containers[i].SecurityContext.Capabilities.Add = append(c.SecurityContext.Capabilities.Add, "SYS_PTRACE")
+					}
+				}
+				pod.Spec.Containers[i].LivenessProbe = nil
+				pod.Spec.Containers[i].ReadinessProbe = nil
+			}
+		}
+	case *corev1.Node:
+		// empty securityContext; uses host namespaces, mounts root partition
+		const volumeName = "host-root"
+		pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
+			Name: volumeName,
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{Path: "/"},
+			},
+		})
+		for i, c := range pod.Spec.Containers {
+			if c.Name == containerName {
+				pod.Spec.Containers[i].VolumeMounts = append(c.VolumeMounts, corev1.VolumeMount{
+					MountPath: "/host",
+					Name:      volumeName,
+				})
+			}
+		}
+		pod.Spec.SecurityContext = nil
+		pod.Spec.HostNetwork = true
+		pod.Spec.HostPID = true
+		pod.Spec.HostIPC = true
+	default:
+		return fmt.Errorf("the %s profile doesn't support objects of type %T", ProfileGeneral, target)
+	}
+
+	return nil
+}
+
+func (p *baselineProfile) Apply(pod *corev1.Pod, containerName string, target runtime.Object) error {
+	switch t := target.(type) {
+	case *corev1.Pod:
+		if t == pod {
+			// For ephemeral container: empty securityContext
+			for i, c := range pod.Spec.EphemeralContainers {
+				if c.Name == containerName {
+					pod.Spec.EphemeralContainers[i].SecurityContext = nil
+				}
+			}
+		} else {
+			// For copy of pod: empty securityContext; sets shareProcessNamespace
+			pod.Spec.SecurityContext = nil
+			pod.Spec.ShareProcessNamespace = pointer.BoolPtr(true)
+			pod.Spec.HostNetwork = false
+			pod.Spec.HostPID = false
+			pod.Spec.HostIPC = false
+			for i, c := range pod.Spec.Containers {
+				if c.Name == containerName {
+					pod.Spec.Containers[i].SecurityContext = nil
+				}
+				pod.Spec.Containers[i].LivenessProbe = nil
```

Comment:
Is it correct to clear probes for all containers in the pod. Shouldn't we do it only for debug container?

Reply:
As far as I can see in here https://github.com/kubernetes/kubernetes/pull/110526/files#diff-789890b76929281f2d5ca2c895ae5294bb6ab24fda7024b31c1db46f452c7c9dR185, probes are cleared for only debug container

Reply:
I think striping every container's probes in the pod copy work better, because in a debug session where you might do something to the network namespace of the pod that causes every container in the pod no longer able to receive or respond to liveness probes, we wouldn't want all the containers to get inadvertently bounced in such a debugging scenario and cause inconvenience. But I'll defer to @verb

Reply:
The original motivation here was that the probes were causing the pod to be killed in the middle of a debugging session.

Since we're stripping the labels so that the pod won't receive traffic it makes sense to strip the probes as well.

At least that's what I remember. :grin: